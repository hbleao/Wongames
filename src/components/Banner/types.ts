export type ImageProps = {
  src: string;
};

export type BannerProps = {
  image: string;
  title: string;
  subtitle: string;
  buttonLabel: string;
  buttonLink: string;
};
