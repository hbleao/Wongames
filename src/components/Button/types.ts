import React, {
  AnchorHTMLAttributes,
  ButtonHTMLAttributes,
  ReactNode
} from 'react';

type Size = 'small' | 'medium' | 'large';
type ButtonTypes =
  | AnchorHTMLAttributes<HTMLAnchorElement>
  | ButtonHTMLAttributes<HTMLButtonElement>;

export type ButtonProps = {
  size?: Size;
  fullWidth?: boolean;
  icon?: ReactNode;
  as?: React.ElementType;
} & ButtonTypes;

export type WrapperProps = {
  fullWidth?: boolean;
  size?: Size;
  hasIcon: boolean;
};
