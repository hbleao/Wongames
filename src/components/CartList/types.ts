import { GameItemProps } from 'components/GameItem/types';

export type CartListProps = {
  items: GameItemProps[];
  total: string;
};
