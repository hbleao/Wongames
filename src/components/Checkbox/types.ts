import { InputHTMLAttributes } from 'react';

export type CheckboxProps = {
  labelText?: string;
  labelFor?: string;
  isChecked?: boolean;
  labelColor?: 'black' | 'white';
  value?: string | ReadonlyArray<string> | number;
  onCheck?: (status: boolean) => void;
} & InputHTMLAttributes<HTMLInputElement>;

export type LabelProps = {
  labelColor?: 'black' | 'white';
};
