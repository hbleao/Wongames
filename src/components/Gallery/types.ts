export type GalleryImageProps = {
  src: string;
  alt: string;
  label: string;
};

export type GalleryProps = {
  items: GalleryImageProps[];
};

export type ModalProps = {
  isOpen: boolean;
};
