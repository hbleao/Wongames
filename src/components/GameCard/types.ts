import { RibbonColors, RibbonSizes } from 'components/Ribbon/types';

export type GameCardProps = {
  title: string;
  developer: string;
  image: string;
  price: string;
  promotionalPrice?: string;
  favorite?: boolean;
  ribbon?: string;
  ribbonSize?: RibbonSizes;
  ribbonColor?: RibbonColors;
  onFav: () => void;
};

export type PriceProps = {
  isPromotional?: boolean;
};
