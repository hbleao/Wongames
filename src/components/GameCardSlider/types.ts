import { GameCardProps } from 'components/GameCard/types';

export type GameCardSliderProps = {
  items: GameCardProps[];
  color?: 'white' | 'black';
};

export type WrapperProps = {
  color?: 'white' | 'black';
};
