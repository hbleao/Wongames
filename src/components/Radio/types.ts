import { InputHTMLAttributes } from 'react';

export type RadioValue = string | ReadonlyArray<string> | number;

export type RadioProps = {
  labelText?: string;
  labelFor?: string;
  isChecked?: boolean;
  labelColor?: 'black' | 'white';
  value?: RadioValue;
  onCheck?: (status?: RadioValue) => void;
} & InputHTMLAttributes<HTMLInputElement>;

export type LabelProps = {
  labelColor?: 'black' | 'white';
};
