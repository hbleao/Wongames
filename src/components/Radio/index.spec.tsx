import React from 'react';
import { screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { Radio } from '.';
import { renderWithTheme } from 'utils/tests/renderWithTheme';

import theme from 'styles/theme';
import { RadioValue } from './types';

export type RadioProps = {
  labelText?: string;
  labelFor?: string;
  labelColor?: 'black' | 'white';
  value?: RadioValue;
  onCheck?: (status?: RadioValue) => void;
};

const makeSut = ({
  labelText,
  labelFor,
  labelColor,
  onCheck,
  value
}: RadioProps) => {
  const sut = renderWithTheme(
    <Radio
      labelText={labelText}
      labelFor={labelFor}
      labelColor={labelColor}
      onCheck={onCheck}
      value={value}
    />
  );

  return {
    sut
  };
};

describe('Components/Radio', () => {
  it('Should be render with label', () => {
    const labelText = 'Radio';
    const labelFor = 'check';
    const { sut } = makeSut({ labelText, labelFor });

    const label = screen.getByLabelText(labelText);
    const radioLabel = screen.getByText(labelText);

    expect(label).toBeInTheDocument();
    expect(radioLabel).toHaveAttribute('for', labelFor);
    expect(sut.container).toMatchSnapshot();
  });

  it('Should be render without label', () => {
    makeSut({});

    const checkboxLabel = screen.queryByText('checkbox');

    expect(checkboxLabel).not.toBeInTheDocument();
  });

  it('Should be render Radio with black label', () => {
    const labelText = 'radio';
    const labelFor = 'check';

    makeSut({ labelText, labelFor, labelColor: 'black' });

    const radioLabel = screen.getByText(labelText);

    expect(radioLabel).toHaveStyle({
      color: theme.colors.black
    });
  });

  it('Should dispatch onCheck when status changes', async () => {
    const labelText = 'radio';
    const labelFor = 'radio';
    const onCheck = jest.fn();

    makeSut({
      labelText,
      labelFor,
      onCheck,
      value: 'any_value'
    });

    expect(onCheck).not.toHaveBeenCalled();

    const radio = screen.getByLabelText('radio');

    userEvent.click(radio);

    await waitFor(() => {
      expect(onCheck).toHaveBeenCalledTimes(1);
    });

    expect(onCheck).toHaveBeenCalledWith('any_value');
  });

  it('Should be accessible with tab', async () => {
    makeSut({ labelFor: 'radio', labelText: 'radio' });

    const body = document.body;
    expect(body).toHaveFocus();

    userEvent.tab();

    const radio = screen.getByLabelText(/radio/i);

    await waitFor(() => {
      expect(radio).toHaveFocus();
    });
  });
});
