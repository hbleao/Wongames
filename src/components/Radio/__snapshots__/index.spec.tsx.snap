// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Radio Should be render with label 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  width: 1.8rem;
  height: 1.8rem;
  border-radius: 50%;
  border: 0.2rem solid #2E2F42;
  background-color: transparent;
  -webkit-transition: background 0.1s ease-in-out;
  transition: background 0.1s ease-in-out;
  outline: none;
  cursor: pointer;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1::before {
  content: '';
  width: 0.8rem;
  height: 0.8rem;
  border-radius: 50%;
  position: absolute;
  opacity: 0;
  -webkit-transition: opacity 0.1s ease-in-out;
  transition: opacity 0.1s ease-in-out;
}

.c1:focus {
  box-shadow: 0 0 0.5rem #F231A5;
}

.c1:checked {
  border-color: #F231A5;
}

.c1:checked:before {
  opacity: 1;
  background-color: #F231A5;
}

.c2 {
  cursor: pointer;
  line-height: 1;
  font-size: 1.6rem;
  padding-left: 0.8rem;
}

<div>
  <div
    class="c0"
  >
    <input
      class="c1"
      id="check"
      type="radio"
      value=""
    />
    <label
      class="c2"
      for="check"
    >
      Radio
    </label>
  </div>
</div>
`;
